### ISSUE TITLE
Artifacts when rendering

### ISSUE THREAD

**Issue by skinkie** [terranodo/tegola#132]
I have just opened https://github.com/mapbox/mapbox-gl-js/issues/5175 to figure out in what part of the server-client the wrong data is send. The config I am testing is:

```
[[providers.layers]]
name = "zones"
geometry_fieldname = "wkb_geometry"
id_fieldname = "ogc_fid"
sql = "SELECT ST_AsBinary(wkb_geometry) AS wkb_geometry, zonenr, zone_id AS ogc_fid, zonenaam, nulzones FROM ovzones_3857 WHERE wkb_geometry && !BBOX!"

[[maps]]
name = "zoning"

[[maps.layers]]
provider_layer = "openov.zones"
min_zoom = 7
max_zoom = 20
```

**Comment by ARolek** [comment]
The artifacts are due to geometry clipping not being enabled on the v0.3.1 version of tegola. You can test geometry clipping by checking out the v0.4.0 branch and compiling from source and enable it by setting the environment variable TEGOLA_CLIPPING="mvt". 

Are you ok with compiling from source? If not I can put together a pre-release build that you can test.

**Comment by skinkie** [comment]
I would be very happy if you could provide a package (and maybe even a bit _more_ happy if I knew how I could solve the dependencies while building from source...) thanks for the effort, it is appreciated.

**Comment by skinkie** [comment]
So for building I did:
```
git checkout v0.4.0
go get github.com/BurntSushi/toml
go get github.com/terranodo/tegola/mvt
go get github.com/terranodo/tegola/mvt/provider
go get github.com/terranodo/tegola/provider/postgis
go get "github.com/terranodo/tegola/server
go get github.com/terranodo/tegola/server
cd cmd/tegola/
rm profile.go
go build -o tegola *.go
```
(minor hacking in main.go and the config stuff that didn't want to compile)

Then I copied tegola to my application folder. And launched via:
```
TEGOLA_CLIPPING="mvt" ./tegola config.toml 
2017/08/21 00:10:31 Loading local config (config.toml)
2017/08/21 00:10:31 Starting tegola server on port :8080
2017/08/21 00:10:42 tile z:12, x:2098, y:1351 is rather large - 533996
```

...I am pretty sure it is not fixed.
![image](https://user-images.githubusercontent.com/502394/29498975-1935c278-8607-11e7-877c-b18b6fb9f917.png)

**Comment by ARolek** [comment]
I'm on mobile and won't be able to build a binary for you until tomorrow. A few things to note:

- you should not need to fetch the dependencies. All dependencies are in the vendor directory and you should not need to go get the tegola packages. Do you clone the repo into your GOPATH? 
- what OS are you using? You mention moving the bin to your application directory but you're also using the ./tegola syntax which indicates you're running the binary from the current directory.

**Comment by skinkie** [comment]
I have never worked with Go, so sadly I have enough to learn. I cloned the server in /usr/src

Arch Linux

**Comment by ARolek** [comment]
@skinkie take a look at the [v0.4.0-alpha release](https://github.com/terranodo/tegola/releases/tag/v0.4.0-alpha) to test clipping support. Make sure to set the environment variable `TEGOLA_CLIPPING=mvt`. Please report back the results.

**Comment by skinkie** [comment]
@ARolek I am very impressed. The hole thing feels so much faster than for example t-rex, and the rendering is without artitifacts.

![impressed](https://user-images.githubusercontent.com/502394/29585065-1894ffd4-8786-11e7-8ea9-a936765bd492.png)

**Comment by skinkie** [comment]
I did find an issue while scrolling through the data. The horizontal while line shouldn't be there.

![artifact](https://user-images.githubusercontent.com/502394/29585187-859aaa70-8786-11e7-80c0-ea55e8b4de42.png)

**Comment by ARolek** [comment]
@skinkie apologies for the delayed response. It's great to hear the experience is much better with the clipping implementation we're working on. There are still a few clipping edge cases that are being worked on as you're seeing here. If you're interested in following the clipping progress issue #56 is where the development is happening. 

It's been quite a large effort to implement all the various pieces necessary to handling clipping correctly within tegola but we're nearing the end of the effort. We're hoping to have our beta release ready soon which also comes with our built in viewer (issue #130) among other improvements.

**Comment by ARolek** [comment]
I'm going to close this issue for now as the remaining artifacts should be addressed once #56 is complete.
